File: C#/Client/ClientProgram/ClientProgram.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net;
using System.Threading.Tasks;
using System.IO;
using System.Net.Sockets;
using System.Threading;

namespace ClientProgram
{
    class ClientProgram
    {
        public static int whoHasBall = -1;
        public static int[] storedIDs = { };

        static void Main(String[] args)
        {
            try
            {
                Thread thread = new Thread(() =>
                {
                    using (Client client = new Client())
                    {
                        Console.WriteLine("Logged in successfully");
                        String line = client.reader.ReadLine();
                        line = client.reader.ReadLine();

                        storedIDs = new int[] { };

                        while (true)
                        {
                            lock (storedIDs)
                            {
                                UpdateGUI(client);
                            }
                        }
                    }
                });
                thread.Start();
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
                Console.WriteLine(e.StackTrace);
            }
        }
        //checks to see if user has ball, if they do they get the option to enter who they want to pass to
        static void UpdateGUI(Client client)
        {
            int currentwhoHasBall = client.whoHasBall();
            int[] ids = client.getPlayersID();
            bool currentPlayerHasBall = client.playerID == currentwhoHasBall;

            if (whoHasBall != currentwhoHasBall || storedIDs.Length != ids.Length)
            {
                Console.WriteLine("Waiting for player: " + currentwhoHasBall + " to pass");
                if (currentPlayerHasBall)
                {
                    Console.WriteLine("You have the ball, select the player you wish to pass to");
                }
            }
            whoHasBall = currentwhoHasBall;
            storedIDs = ids;
        }
    }
}

File: Java/Client/src/ClientProgram.java
//package ;

import javax.swing.*;
import java.util.Scanner;

public class ClientProgram {
    public static ClientGUI clientGUI;
    public static int whoHasBall;
    public static int[] storedIDs;

    public static void main(String[] args) {
        try {
            //creates a client object with its GUI and reads from the server what its id is
            Scanner in = new Scanner(System.in);
            Client client = new Client();
            System.out.println("Logged in successfully");
            client.reader.nextLine();
            client.reader.nextLine();
            clientGUI = new ClientGUI(client);

            new Thread(() -> {
                while (true) {
                    updateGUI(client);
                }
            }).start();
        } catch (Exception e) {

            System.out.println(e.getMessage());
            e.printStackTrace();
        }
    }
    //updates GUI and enables the pass button if the client has the ball
    public static void updateGUI(Client client) {

        int currentwhoHasBall = client.whoHasBall();
        int[] ids = client.getPlayersID();
        boolean currentPlayerHasBall = client.playerID == currentwhoHasBall;
        if (whoHasBall != currentwhoHasBall || storedIDs.length != ids.length) {

            SwingUtilities.invokeLater(() -> {
                clientGUI.label.setText("Waiting for player: " + currentwhoHasBall + " to pass");
                clientGUI.button.setEnabled(false);
                clientGUI.dropDown.removeAllItems();
                for (int id : ids) {
                    String inputString = "Player: " + id;
                    clientGUI.dropDown.addItem(inputString);
                }
                if (currentPlayerHasBall) {
                    clientGUI.label.setText("You have the ball, select the player you wish to pass to");
                    clientGUI.button.setEnabled(true);
                }
            });
            whoHasBall = currentwhoHasBall;
            storedIDs = ids;
        }
    }
}
